### DISCUSSION TITLE
How to use cv2 and toga

### DISCUSSION THREAD

**Discussion by paolovertullo** [beeware/briefcase#1921] [Q&A]
Hi everyone,
Im looking for an example that use toga and cv2 to get webcam stream to apply a tflite object detection and tracking.  I already have a model that works fine but I cant manage to use toga. My goal is to deploy my app on android.

Do you know any example or project that has those features ?

Paolo

**Comment by freakboy3742** [comment]
In general, there's no special instructions for how to use *any* package in Briefcase. Add the package to your requirement, and run either `briefcase create` (for a new project) or `briefcase update -r` to update requirements on an existing project.

The caveat on that is that the package needs to *exist* for the platform you want to target. In your case, you want to target Android and use the camera APIs, so you need a version of the opencv wheel that has been compiled for Android with camera support enabled. At present, [opencv-python is on the list of packages that is currently supported](https://chaquo.com/pypi-13.1/opencv-python/), but [the camera APIs haven't been enabled](https://github.com/chaquo/chaquopy/issues/620). Apparently OpenCV *has* Android support for camera APIs; but that support hasn't been enabled as part of the compiled wheel. This is likely solvable, but it requires someone looking into the problem.

**Reply by mhsmith** [reply]
As a possible workaround, you could use [Toga's Camera API](https://toga.readthedocs.io/en/stable/reference/api/hardware/camera.html), though it currently only supports still images, not video.
